jobs/integration/validation.py

```diff
@@ -574,9 +574,14 @@ async def get_to_restricted_networkpolicy_service():
 
 
 @pytest.mark.asyncio
-@pytest.mark.skip_apps(["calico"])
 async def test_ipv6(model, tools):
-    master = model.applications["kubernetes-master"].units[0]
+    master_app = model.applications["kubernetes-master"]
+    master_config = await master_app.get_config()
+    service_cidr = master_config["service-cidr"]["value"]
+    if all(ipaddress.ip_network(cidr).version != 6 for cidr in service_cidr.split(',')):
+        pytest.skip("kubernetes-master not configured for IPv6")
```

Comment:
Calico doesn't support IPv6 + VXLAN, so we have to disable IPv6 when testing VXLAN. I had to make the test_ipv6 skip logic a bit smarter to accomodate that.